operator/controllers/kyma_controller.go

```diff
@@ -575,29 +592,42 @@ func (r *KymaReconciler) DeleteKymaDependencies(ctx context.Context, kyma *opera
 	return nil
 }
 
-func (r *KymaReconciler) UpdateStatusAndHandleErr(
+func (r *KymaReconciler) UpdateStatus(
 	ctx context.Context, kyma *operatorv1alpha1.Kyma, state operatorv1alpha1.KymaState, message string,
 ) error {
 	if err := status.Helper(r).UpdateStatus(ctx, kyma,
-		operatorv1alpha1.KymaStateError, "templates could not be fetched"); err != nil {
+		state, "templates could not be fetched"); err != nil {
 		return fmt.Errorf("error while updating status to %s because of %s: %w", state, message, err)
 	}
+	r.Event(kyma, "Normal", "StatusUpdate", message)
+	return nil
+}
+
+func (r *KymaReconciler) UpdateStatusFromErr(
+	ctx context.Context, kyma *operatorv1alpha1.Kyma, state operatorv1alpha1.KymaState, err error,
+) error {
+	if err := status.Helper(r).UpdateStatus(ctx, kyma,
+		state, err.Error()); err != nil {
+		return fmt.Errorf("error while updating status to %s: %w", state, err)
+	}
+	r.Event(kyma, "Warning", "StatusUpdate", err.Error())
 	return nil
 }
 
 func (r *KymaReconciler) GetModules(
 	ctx context.Context, kyma *operatorv1alpha1.Kyma, checkOutdated bool,
-) (util.Modules, error) {
+) (util.ParsedModules, error) {
 	// fetch templates
 	templates, err := release.GetTemplates(ctx, r, kyma)
 	if err != nil {
-		return nil, r.UpdateStatusAndHandleErr(ctx, kyma, operatorv1alpha1.KymaStateError, "templates could not be fetched")
+		return nil, fmt.Errorf("templates could not be fetched: %w", err)
```

Comment:
we don't update the kyma state in this case?

Reply:
I refactored it so that GeModules always gets called as lower level method. The Method is called from places that always update the status on error, so it was redundant